frontend/app/assets/javascripts/spree/frontend/views/spree/products/index.js

```diff
@@ -17,16 +17,35 @@ Spree.ready(function($) {
     $(this).toggleClass('plp-overlay-card-item--selected')
   })
 
-  $('.color-select').click(function() {
-    var allOptionsBorders = $('.color-select-border')
-    var borderElement = $(this).find('.color-select-border')
+  var allOptionsBorders = $('.color-select-border')
+  var addToCartButton = document.getElementById('add-to-cart-button')
 
+  function removeSelectedBorders() {
     allOptionsBorders.each(function() {
       $(this).removeClass('color-select-border--selected')
     })
+  }
+
+  $('.color-select').click(function() {
+    var borderElement = $(this).find('.color-select-border')
+
+    removeSelectedBorders()
     borderElement.addClass('color-select-border--selected')
   })
 
+  if (allOptionsBorders !== undefined && addToCartButton !== null) {
```

Comment:
`allOptionsBorders !== undefined`
will not work with jquery dom element. Change to this `allOptionsBorders.length`